torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2730,6 +2784,34 @@ def reference_sgn(x):
                    dtypesIfCPU=all_types_and_complex_and(torch.half, torch.bfloat16),
                    dtypesIfCUDA=all_types_and_complex_and(torch.half, torch.bfloat16),
                    assert_autodiffed=True,),
+    OpInfo('pow',
+           dtypes=all_types_and_complex_and(torch.half, torch.bfloat16),
+           sample_inputs_func=sample_inputs_pow,
+           supports_inplace_autograd=False,
+           assert_autodiffed=True,
+           skips=(
+               # Inplace variants currently use the same sample inputs as the method for which
+               # they are a variant, causing this test to fail because of broadcasting semantics.
+               # This test should be enabled after GitHub PR 53014 would land.
+               # Reference: https://github.com/pytorch/pytorch/issues/50747
+               SkipInfo('TestCommon', 'test_variant_consistency_eager',
+                        dtypes=[torch.float32, torch.float64]),
+               # Due to AVX2 curently not being fully supported for Float16, log_vml_cpu can't be enabled
+               # for Float16, causing this test to fail.
+               SkipInfo('TestOpInfo', 'test_supported_backward',
```

Comment:
Could you elaborate more on this please? Does this mean autograd is not supported for Half dtype? If so, do we want to enable a dtype with incomplete support?

Reply:
This test was only yesterday. Earlier, `test_variant_consistency_jit` had been failing due to this reason, but now that test isn't being run for `float16`. `test_variant_consistency_eager` had been passing for float16, though, so it seemed to me that `pow` wasn't _jittable_ for `Float16`. But does that failure mean that even eager mode is not supported?

Reply:
I looked up old CI logs. If no sample inputs are used that cause trouble due to broadcasting semantics for inplace variants, then `test_variant_consistency_eager` did use to pass on `float16` for `pow`, so backward does seem to be supported for it in eager mode.

So I'm very confused.

Reply:
[mruberry helped me understand](https://github.com/pytorch/pytorch/pull/53255#issuecomment-805477396) that `pow` doesn't indeed support `backward` with `float16`.
Should `pow` still be enabled for `float16` for use-cases that don't require autograd?

Or should it be deferred to whenever more AVX-2 support would be added for `float16`?

Reply:
This is a good question, but yes we should enable it even if it's forward-only. This will enable backward for other operations that depend on pow in their backwards operations, for example. In the future we can look at adding autograd support for pow in fp16, too.

Also in the future we should probably refine our docs to include these properties, too. But both the doc improvement and autograd support are tasks for later, and there are probably higher priority things to do now.

Reply:
I agree we should have a more consistent way to document supported features for operators (e.g. autograd, dtypes, broadcasting, etc).